sqlx-core/src/url.rs

```diff
@@ -52,8 +52,17 @@ impl Url {
         }
     }
 
-    pub fn password(&self) -> Option<&str> {
-        self.0.password()
+    pub fn password(&self) -> Option<Cow<str>> {
+        match self.0.password() {
+            Some(s) => {
+                let decoded = percent_encoding::percent_decode_str(s);
+
+                // Note: `unwrap()` here can't fail since we started with valid
+                // UTF-8 to begin with
+                Some(decoded.decode_utf8().unwrap())
```

Comment:
It's actually possible to percent-encode binary data which would cause a panic here. Though I'm unsure what we should do in practice. It may be a good idea to follow diesel here and have a `BadUrl` or `InvalidUrl` error variant. We need to have a more general discussion of our errors though.. so I don't think you should fix this now. For now, can you just adjust the comment like `// FIXME: Handle error` or something and we can look at it when we fix up our errors.

Reply:
Ah true, good catch :) adjusted as suggested!

Reply:
Actually I would change this to `.expect("percent-encoded password contained non-UTF-8 bytes")` or else the user won't easily find what caused the panic unless they have backtraces on.

Reply:
yup that sounds fair to me